tfphasespace/tfphasespace.py

```diff
@@ -25,117 +25,201 @@
 
 
 def process_list_to_tensor(lst):
+    """Convert a list to a tensor.
+
+    The list is converted to a tensor and transposed to get the proper shape.
+
+    Note:
+        If `lst` is a tensor, nothing is done to it other than convert it to
+        `tf.float64`.
+
+    Arguments:
+        lst (list): List to convert.
+
+    Return:
+        Tensor.
+
+    """
     if isinstance(lst, list):
         lst = tf.transpose(tf.convert_to_tensor(lst,
                                                 preferred_dtype=tf.float64))
     return tf.cast(lst, dtype=tf.float64)
 
 
 def pdk(a, b, c):
-    """Calculate the PDK (2-body phase space) function."""
+    """Calculate the PDK (2-body phase space) function.
+
+    Based on Eq. (9.17) in CERN 68-15 (1968).
+
+    Arguments:
+        a, b, c (Tensor): M_{i+1}, M_{i}, m_{i+1} in Eq. (9.17).
+
+    Return:
+        Tensor.
+
+    """
     x = (a - b - c) * (a + b + c) * (a - b + c) * (a + b - c)
     return tf.sqrt(x) / (_TWO * a)
 
 
 class Particle:
-    def __init__(self, name, parent=None):
+    """Representation of a particle.
+
+    Instances of this class can be combined with each other to build decay chains,
+    which can then be used to generate phase space events through the `generate`
+    method.
+
+    A `Particle` must have a `name`, which is ensured not to clash with any others in
+    the decay chain.
+    It may also have:
+        + Mass, which can be either a number or a function to generate it according to
+        a certain distribution. In this case, the particle is not considered as having a
+        fixed mass and the `has_fixed_mass` method will return False.
+        + Children, ie, decay products, which are also `Particle` instances.
+
+    """
```

Comment:
Probably move that to `__init__` so that we can document the parameters as well?

Reply:
reconsidering it's good as it is, sorry